core/src/main/java/com/scalar/db/api/TableMetadata.java

```diff
@@ -169,4 +169,44 @@ public TableMetadata build() {
           columns, partitionKeyNames, clusteringKeyNames, clusteringOrders, secondaryIndexNames);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    TableMetadata metadata = (TableMetadata) o;
+
+    if (!Objects.equals(columnNames, metadata.columnNames)) {
+      return false;
+    }
+    if (!Objects.equals(columnDataTypes, metadata.columnDataTypes)) {
+      return false;
+    }
+    if (!Objects.equals(partitionKeyNames, metadata.partitionKeyNames)) {
+      return false;
+    }
+    if (!Objects.equals(clusteringKeyNames, metadata.clusteringKeyNames)) {
+      return false;
+    }
+    if (!Objects.equals(clusteringOrders, metadata.clusteringOrders)) {
+      return false;
+    }
+    return Objects.equals(secondaryIndexNames, metadata.secondaryIndexNames);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = columnNames != null ? columnNames.hashCode() : 0;
```

Comment:
I think we should use Objects.hash() or guava's Objects.hashCode() instead of re-implement the hashing.
https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained#hashcode